presentation/src/main/java/com/knocklock/presentation/lockscreen/receiver/StartApplicationReceiver.kt

```diff
@@ -0,0 +1,48 @@
+package com.knocklock.presentation.lockscreen.receiver
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.util.Log
+import dagger.hilt.android.AndroidEntryPoint
+
+/**
+ * @Created by 김현국 2022/12/13
+ * @Time 4:00 PM
+ */
+
+@AndroidEntryPoint
+class StartApplicationReceiver(
+    private val context: Context,
+    private val openLockScreen: OpenLockScreen
+) : BroadcastReceiver() {
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        startLockScreen(context)
+    }
+
+    private fun startLockScreen(context: Context?) {
+        if (context != null) {
+            try {
+                openLockScreen.openLockScreenByIntent()
+            } catch (e: Exception) {
+                Log.e("Receiver Exception", e.stackTraceToString())
+            }
+        }
+    }
+    fun stop() {
+        context.unregisterReceiver(this)
+    }
+
+    fun start() {
+        val intentFilter = IntentFilter().apply {
+            addAction(Intent.ACTION_SCREEN_OFF)
+            addAction(Intent.ACTION_BOOT_COMPLETED)
+        }
+        context.registerReceiver(this, intentFilter)
+    }
+}
+interface OpenLockScreen {
```

Comment:
Callback interface 형태는 Listener로 네이밍을 통일하는 것도 좋아보여요!
```suggestion
interface OnOpenLockScreenListener {
```

Reply:
네네  Listener로 통일화할께용

Reply:
@jaeryo2357  [2b3fe7d](https://github.com/Knock-Lock/Knock-Lock/pull/69/commits/2b3fe7dec11fea11c2c863e30ac3e8f89990cb10)
[6a665e8](https://github.com/Knock-Lock/Knock-Lock/pull/69/commits/6a665e8a3ecce6e49d481a7c75e133be02c63c9d) 에서 반영했습니당